src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -17,13 +17,12 @@
 
 namespace Xamarin.Android.Tasks
 {
-	public class MonoAndroidHelper
+	public partial class MonoAndroidHelper
 	{
 		static Lazy<string> uname = new Lazy<string> (GetOSBinDirName, System.Threading.LazyThreadSafetyMode.PublicationOnly);
 
 		// Set in ResolveSdks.Execute();
 		// Requires that ResolveSdks.Execute() run before anything else
-		public static string[] TargetFrameworkDirectories;
```

Comment:
Does anything set this when the linker is running? I would guess it lives in a different `static` scope than the rest of XA's build?

Reply:
Nothing sets it when used in ILLlink.

It is used through `IsFrameworkAssembly` in `MarkJavaObjects`. Maybe we can replace the `IsFrameworkAssembly` with `IsSdk`, `IsProduct`  calls?

Reply:
In the end I set it in the setup step, passing the paths through the `--custom-data`